zigzag-circuit.md

```diff
@@ -49,37 +49,37 @@ This circuit proves that given a Merkle root `CommD`, `CommR_l`, and `commRStar`
 
```

Comment:
It think `ParentInclusionHash` and `ParentInclusionLeaf` should have `{0..EXPANSION_DEGREE+BASE_DEGREE}`.





Also, if we're separating the leaves from the paths, shouldn't those paths  be shorter by 1 — so `TREE_DEPTH-2`?

Related: Below it looks like we are using `{0..TREE_DEPTH}`  in `MerkleTreeVerify`. It should be the same as here (whatever that ends up being).

Reply:
@nicola

Reply:
1) `ParentInclusionHash` and `ParentInclusionPaths` have `{0..EXPANSION_DEGREE+BASE_DEGREE}`, what do you mean exactly?
2) TREE_DEPTH-1 is including the path being shorter by 1
3) Thanks will update!

Reply:
Sorry, I think Github hid the context. See lines 45 and 48.

45: `ParentInclusionHash_{l=0..LAYERS}_{c=0..LAYER_CHALLENGES[i]}_{EXPANSION_DEGREE+ BASE_DEGREE}_{0..TREE_DEPTH-1}`

The part with `{EXPANSION_DEGREE+BASE_DEGREE}` isn't specifying a range in our syntax. It should be `{0..EXPANSION_DEGREE +BASE_DEGREE}`.

Reply:
2. Maybe we are defining depth differently.

I consider a tree with four nodes to have depth 3. It has three rows with 1, 2, and 4 nodes each.

So an inclusion proof starting at the leaves only needs 3-1=2 path bits and elements. That is, the whole path is TREE_DEPTH-1 in length. That path is composed of 1 leaf and TREE_DEPTH-2 hashes.

I could be seeing this wrong or getting the terminology wrong, but that is what I meant.

Reply:
solved point 1, on point 2: TREE_DEPTH-1 means we don't hash the leaf, it would be TREE_DEPTH otherwise (or am I missing something?)

Reply:
No, I see it now. I was thinking that since we reuse the root we don't have to hash it, but actually we still have to hash the previous row to compute the root. So I think everything is resolved now.